backend/src/ml_space_lambda/dataset/lambda_functions.py

```diff
@@ -258,35 +258,85 @@ def list_locations(event, context):
 
 @api_wrapper
 def list_files(event, context):
-    prefix = get_dataset_prefix(
+    env_variables = get_environment_variables()
+    query_string_parameters = event["queryStringParameters"] or {}
+    
+    # map query parameters keys to api parameter names
+    query_string_parameters = map_dict_keys(query_string_parameters, {
+        "nextToken": "ContinuationToken",
+        "pageSize": "MaxKeys",
+        "prefix": "Prefix"
+    })
+    
+    dataset_prefix = get_dataset_prefix(
         event["pathParameters"]["scope"], event["pathParameters"]["datasetName"]
     )
-
-    env_variables = get_environment_variables()
-    keys = []
-
-    if event["queryStringParameters"]:
-        s3_response = s3.list_objects_v2(
-            Bucket=env_variables["DATA_BUCKET"],
-            Prefix=prefix,
-            ContinuationToken=event["queryStringParameters"].get("nextToken", ""),
-        )
-    else:
-        s3_response = s3.list_objects_v2(Bucket=env_variables["DATA_BUCKET"], Prefix=prefix)
-
-    for object in s3_response.get("Contents", []):
-        keys.append({"key": object["Key"], "size": object["Size"]})
-
-    # Note: For s3_response if there are no files, the path to the dataset is returned
-    # Remove the dataset path to correctly reflect there are no files in the dataset
-    if len(keys) == 1 and keys[0]["key"] == prefix:
-        keys = []
-
-    response = {
-        "Keys": keys,
+    computed_prefix = "".join([dataset_prefix, query_string_parameters.get("Prefix", "")])
+    
+    query_parameters = {
+        "Bucket": env_variables["DATA_BUCKET"],
+        "Prefix": computed_prefix,
+        "Delimiter": "/"
     }
-
-    if "NextContinuationToken" in s3_response:
-        response["nextToken"] = s3_response["NextContinuationToken"]
-
-    return response
+    
+    # constrain which query parameters can be supplied
+    allowed_query_string_parameters = ["MaxKeys", "ContinuationToken"]
+    for key in filter_dict(query_string_parameters, allowed_query_string_parameters):
+        query_parameters[key] = query_string_parameters[key]
+
+    s3_response = s3.list_objects_v2(**query_parameters)
+
+    # drop unwated fields from response
+    preserved_response_keys = ["IsTruncated", "MaxKeys", "Prefix", "NextContinuationToken"]
```

Comment:
Do we actually need `IsTruncated` if we have `NextContinuationToken`? Can we infer this based on the existence of `NextContinuationToken`? Not a big deal since we're just passing through the results of the API call but curious what we'll be using it for.

Reply:
I don't _think_ so. I was torn on if to include it or not.